code/script.js

```diff
@@ -1,47 +1,247 @@
+    showMessage(wineChoice, 'user');
+    wineInput.value = '';
+    setTimeout(() => validateResponse(wineChoice), 1000);
+};
+
+
+// 3   The user's wine choice is validated and bot responds based on user's input
+const validateResponse = (wineChoice) => {
+
+    if (wineChoice.toLowerCase() === 'red') {
+        showMessage(`${wineChoice} is it?`, 'bot');
+        setTimeout(() => questionRed(), 2000);
+    } else if (wineChoice.toLowerCase() === 'white') {
+        showMessage(`${wineChoice}... Are you sure?`, 'bot');
+        setTimeout(() => questionWhite(), 2000);
+    } else {
+        showMessage(`Please answer if you prefer red or white!`, 'bot');
+    }
+};
+
+
+// RED WINE CONVERSATION
+// 4   Red wine - user chooses the preferred wine country
+const questionRed = () => {
+    showMessage(`Great choice! Which of the following countries would you say that you prefer when it comes to red wine?`, 'bot');
+
+    inputWrapper.innerHTML = `
+        <button id="italy">Italy</button>
+        <button id="spain">Spain</button>
+    `
+
+    document.getElementById('italy').addEventListener('click', () => {
+        showMessage('Italy', 'user');
+        setTimeout(() => questionGrapeItaly(), 2000);
+    })
+
+
+    document.getElementById('spain').addEventListener('click', () => {
+        showMessage('Spain', 'user');
+        setTimeout(() => questionGrapeSpain(), 2000);
+    })
+};
+
+
+// 4.1    Italy - user chooses the preferred grape from Italy
+const questionGrapeItaly = () => {
+    showMessage(`Perfect! Might as well! Which of the following grapes do you prefer?`, 'bot');
+
+    inputWrapper.innerHTML = `
+        <button id="neroDavola">Nero d'Avola</button>
+        <button id="barbera">Barbera</button>
+        <button id="amarone">Amarone</button>
+    `
+
+    document.getElementById('neroDavola').addEventListener('click', () => {
+        showMessage("Nero d'Avola", 'user');
+        setTimeout(() => wineRecoNero(), 2000);
+    })
+
+    document.getElementById('barbera').addEventListener('click', () => {
+        showMessage('Barbera', 'user');
+        setTimeout(() => wineRecoBarbera(), 2000);
+    })
+
+    document.getElementById('amarone').addEventListener('click', () => {
+        showMessage('Amarone', 'user');
+        setTimeout(() => wineRecoAmarone(), 2000);
+    })
+};
+
+
+// 4.1.1    Italy - bot gives the last message based on user's choice on grape
+const wineRecoNero = () => {
+    showMessage(`Fantastic choice! The finest Nero d'Avola can be found on Sicily! Enjoy!`, 'bot');
+    sound.play();
+    inputWrapper.innerHTML = ``;
+}
+
+const wineRecoBarbera = () => {
+    showMessage(`An amazing grape! It's the most planted grape in all of Italy! Enjoy!`, 'bot');
+    sound.play();
+    inputWrapper.innerHTML = ``;
+}
+
+const wineRecoAmarone = () => {
+    showMessage(`Exclusive taste, eh? It's worth every penny! Enjoy!`, 'bot');
+    sound.play();
+    inputWrapper.innerHTML = ``;
+}
```

Comment:
Those functions are in fact the same. Try to tweak it a little bit to have only one?